### DISCUSSION TITLE
Welcome to topshot-sales Discussions!

### DISCUSSION THREAD

**Discussion by rrrkren** [rrrkren/topshot-sales#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Nhandos** [comment]
# [Question] Retrieve playID & metadata with momentID

So, as far as I can understand - the only ways to map a "momentID" to "playID" when polling MomentsPurchased events are to:

1. Poll all MomentMintedEvents which contains the momentID, playID, & setID from the beginning of the block chain
2. Look up the sales collection of the seller at the previous block by using ExecuteScriptAtBlockHeight()

The limitation of method 1 is that events are archived past a certain N blocks (about 30 days when I test with my script). And the limitation of method 2 is that you can't execute the script at any arbitrary block height.

Am I correct? I just want to know how CryptoSlam.io does it, you can pull all the metadata (playID, setID, playmetadata etc...) by giving a momentID
https://www.cryptoslam.io/nba-top-shot/mint/582211

**Comment by aaronlinsky** [comment]
@rrrkren Thanks for putting this repo out there. 

Is there any way to retrieve/query archived events?

**Comment by hoffmanroni** [comment]
@rrrkren thanks for this.  Is there some way to get the date/time of the MomentPurchase event?  I just see price and seller but is there way to know when the purchase took place?

Thanks!

**Comment by alvins82** [comment]
Anyone have any ideas on how to go from a MomentPurchased to a URL for the moment? I can see cryptoslam has links to some moments but not all.
